lib/matplotlib/tests/test_contour.py

```diff
@@ -143,6 +143,25 @@ def test_contour_empty_levels():
     assert len(record) == 1
 
 
+def test_contour_badlevel_fmt():
+    # test funny edge case from
+    # https://github.com/matplotlib/matplotlib/issues/9742
+    # User supplied fmt for each level as a dictionary, but
+    # MPL changed the level to the minimum data value because
+    # no contours possible.
+    # This would error out pre
+    # https://github.com/matplotlib/matplotlib/pull/9743
+    x = np.arange(9)
+    z = np.zeros((9, 9))
+
+    fig, ax = plt.subplots()
+    fmt = {1.: '%1.2f'}
+    with pytest.warns(UserWarning) as record:
+        cs = ax.contour(x, x, z, levels=[1.])
+        ax.clabel(cs, fmt=fmt)
+    assert len(record) == 1
```

Comment:
Shouldn't this be inside the `with` block?

Reply:
I just copied the previous test.   Seems to work but happy to change if wrong

Reply:
OK, I see now where this style is shown in the pytest documentation, so no problem.